B-20293: MTO Page Shipment Header Missing Values

TevinAdams opened the pull request:
## [Agility ticket](https://www13.v1host.com/USTRANSCOM38/assetdetail.v1?number=B-20293)

## Summary

The location and date information under the shipment header is being removed as requested in the feature this BL item is under.

[this article](tbd) explains more about the approach used.

## Verification Steps for the Author

These are to be checked by the author.

- [ ] Tested in the Experimental environment (for changes to containers, app startup, or connection to data stores)
- [ ] Have the Agility acceptance criteria been met for this change?

## Verification Steps for Reviewers

These are to be checked by a reviewer.

- [ ] Has the branch been pulled in and checked out?
- [ ] Have the BL acceptance criteria been met for this change?
- [ ] Was the CircleCI build successful?
- [ ] Has the code been reviewed from a standards and best practices point of view?

### Setup to Run the Code

- [Instructions for starting storybook](https://transcom.github.io/mymove-docs/docs/frontend/setup/storybook)
- [Instructions for starting the MilMove application](https://transcom.github.io/mymove-docs/docs/getting-started/application-setup/)
- [Instructions for running tests](https://transcom.github.io/mymove-docs/docs/getting-started/development/testing)

### How to test

1. Access office
2. Login as a TOO, TIO or counselor
3.Navigate select a move with a PPM or any other shipment
4. Navigate to the MTO page and examine a shipment
5. Notice the Shipment address information should not be there

### Frontend

- [ ] There are no aXe warnings for UI.
- [ ] This works in [Supported Browsers and their phone views](https://transcom.github.io/mymove-docs/docs/adrs/Browser-Support/#minimum-browser-requirements) (Chrome, Firefox, Edge).
- [ ] There are no new console errors in the browser devtools.
- [ ] There are no new console errors in the test output.
- [ ] If this PR adds a new component to Storybook, it ensures the component is fully responsive, OR if it is intentionally not, a wrapping div using the `officeApp` class or custom `min-width` styling is used to hide any states the would not be visible to the user.
- [ ] This change meets the standards for [Section 508 compliance](https://www.ssa.gov/accessibility/andi/help/install.html).

### Backend

- [ ] Code follows the guidelines for [Logging](https://transcom.github.io/mymove-docs/docs/getting-started/development/logging).
- [ ] The requirements listed in [Querying the Database Safely](https://transcom.github.io/mymove-docs/docs/backend/guides/golang-guide#querying-the-database-safely) have been satisfied.

### Database

#### Any new migrations/schema changes:

- [ ] Follows our guidelines for [Zero-Downtime Deploys](https://transcom.github.io/mymove-docs/docs/backend/setup/database-migrations#zero-downtime-migrations).
- [ ] Have been communicated to #g-database.
- [ ] Secure migrations have been tested following the instructions in our [docs](https://transcom.github.io/mymove-docs/docs/backend/setup/database-migrations#secure-migrations).

## Screenshots
Before:
![image](https://github.com/user-attachments/assets/3d877b16-090c-4a0f-af2e-f283a6734152)

After: 
![image](https://github.com/user-attachments/assets/f241ae9e-4623-40d5-941e-a9479a7f7c88)

508:
*Note the header was removed
![image](https://github.com/user-attachments/assets/00be8c65-250b-40c3-848a-346543a9f5c5)

danieljordan-caci commented:
Looks like you have a failing client test

danieljordan-caci commented:
> Looks like you have a failing client test

Disregard this - ain't on you

traskowskycaci reviewed (changes requested):
Tested by creating a new move and counseling it then viewing as the TOO - not seeing the address show up

Added a few console logs - you might need to dig into what populates the origin information on that shipment status

![image](https://github.com/user-attachments/assets/ec3265ee-4b90-46f1-b9c0-d8dd333ab7e7)


![image](https://github.com/user-attachments/assets/bb7ee1cd-a537-420a-b671-38fa2f3a4950)

TevinAdams commented:
> Tested by creating a new move and counseling it then viewing as the TOO - not seeing the address show up
> 
> Added a few console logs - you might need to dig into what populates the origin information on that shipment status
> 
> ![image](https://private-user-images.githubusercontent.com/147537467/359266589-ec3265ee-4b90-46f1-b9c0-d8dd333ab7e7.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjQxMDQzMDEsIm5iZiI6MTcyNDEwNDAwMSwicGF0aCI6Ii8xNDc1Mzc0NjcvMzU5MjY2NTg5LWVjMzI2NWVlLTRiOTAtNDZmMS1iOWMwLWQ4ZGQzMzNhYjdlNy5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwODE5JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDgxOVQyMTQ2NDFaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1iOTZiMTgxZmE5OWYyMDU0MWMzOWExNTMzNWY1MWU3OTlkMDY5NGU2MzZhNGNkNmI1MzU3NjU0ZjkwOWZiOWUwJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.2To0r3C7uNJX23PJ0nLrh5tDrMu9LHj9zHFKq0LHEBg)
> 
> ![image](https://private-user-images.githubusercontent.com/147537467/359266730-bb7ee1cd-a537-420a-b671-38fa2f3a4950.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjQxMDQzMDEsIm5iZiI6MTcyNDEwNDAwMSwicGF0aCI6Ii8xNDc1Mzc0NjcvMzU5MjY2NzMwLWJiN2VlMWNkLWE1MzctNDIwYS1iNjcxLTM4ZmEyZjNhNDk1MC5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwODE5JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDgxOVQyMTQ2NDFaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1mNGE5ZDQyOWZiNTUwOWEwZjZlZTc4N2UwZTgwZWU3ZmQzZTczMjc3YWI4MDZiNzc3ZTMxZTQ1ZmZjMmVkNGVmJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9._NJJD0YQh9R8coxUK09Lhq0Vd8GWbwZHTo9Azo2leRQ)

@traskowskycaci This one is missing the originPostalCode, originCity and originState. In this case the header should not show up because it would be missing information. If you go to the DB and look at the shipment, populate those values and you will see the header show up.

traskowskycaci commented:
> @traskowskycaci This one is missing the originPostalCode, originCity and originState. In this case the header should not show up because it would be missing information. If you go to the DB and look at the shipment, populate those values and you will see the header show up.

Without modifying the db, how should I populate those values?

traskowskycaci commented:
> > @traskowskycaci This one is missing the originPostalCode, originCity and originState. In this case the header should not show up because it would be missing information. If you go to the DB and look at the shipment, populate those values and you will see the header show up.
> 
> Without modifying the db, how should I populate those values?

Only asking because I want to make sure we are actually populating those originPostalCode, originCity and originState somewhere during the process and not just filling them out in the DB @TevinAdams

danieljordan-caci commented on src/components/Office/ShipmentHeading/ShipmentHeading.jsx:
```diff
@@ -26,6 +26,12 @@ function ShipmentHeading({ shipmentInfo, handleShowCancellationModal, isMoveLock
   const showRequestCancellation =
     shipmentStatus !== shipmentStatuses.CANCELED && shipmentStatus !== shipmentStatuses.CANCELLATION_REQUESTED;
   const isCancellationRequested = shipmentStatus === shipmentStatuses.CANCELLATION_REQUESTED;
+  const showShipmentAddressInfo =
+    shipmentInfo?.originCity &&
+    shipmentInfo?.originState &&
+    shipmentInfo?.originPostalCode &&
+    shipmentInfo?.destinationAddress &&
+    shipmentInfo?.scheduledPickupDate;
```
In `MoveTaskOrder.jsx:1210` you can see what is passed in for these values:
```
shipmentInfo={{
                    shipmentID: mtoShipment.id,
                    shipmentType: mtoShipmentTypes[mtoShipment.shipmentType],
                    isDiversion: mtoShipment.diversion,
                    originCity: pickupAddress?.city || '',
                    originState: pickupAddress?.state || '',
                    originPostalCode: pickupAddress?.postalCode || '',
                    destinationAddress: destinationAddress || dutyLocationPostal,
                    scheduledPickupDate: formattedScheduledPickup,
                    shipmentStatus: mtoShipment.status,
                    ifMatchEtag: mtoShipment.eTag,
                    moveTaskOrderID: mtoShipment.moveTaskOrderID,
                    shipmentLocator: mtoShipment.shipmentLocator,
                  }}
```

The problem is PPM shipments don't use the `pickupAddress` value on the `mtoShipment` level. It has a nested object inside `mtoShipment` where it stores that information. 

`mtoShipment.ppmShipment`

Screenshot here of the object (logging `mtoShipment` on `MoveTaskOrder.jsx` prior to `ShipmentHeading`:
![Screenshot 2024-08-20 at 9 06 31 AM](https://github.com/user-attachments/assets/fdcdfddb-35bb-489a-90e3-7af3167eb667)

You need to dynamically render either the `mtoShipment.pickupAddress` || `mtoShipment.destinationAddress` or the `mtoShipment.ppmShipment.pickupAddress` || `mtoShipment.ppmShipment.destinationAddress` at the MTO level - not the ShipmentHeading component level.

TevinAdams replied:
Done

danieljordan-caci commented on src/pages/Office/MoveTaskOrder/MoveTaskOrder.jsx:
```diff
@@ -1212,10 +1213,16 @@ export const MoveTaskOrder = (props) => {
-                    originState: pickupAddress?.state || '',
-                    originPostalCode: pickupAddress?.postalCode || '',
-                    destinationAddress: destinationAddress || dutyLocationPostal,
+                    originCity: isPPM ? mtoShipment?.ppmShipment?.pickupAddress?.city || '' : pickupAddress?.city || '',
+                    originState: isPPM
+                      ? mtoShipment?.ppmShipment?.pickupAddress?.state || ''
+                      : pickupAddress?.state || '',
+                    originPostalCode: isPPM
+                      ? mtoShipment?.ppmShipment?.pickupAddress?.postalCode || ''
+                      : pickupAddress?.postalCode || '',
+                    destinationAddress: isPPM
+                      ? mtoShipment?.ppmShipment?.destinationAddress || ''
+                      : destinationAddress || dutyLocationPostal,
```
Can I make a request to clean up the props a little more?

A suggestion would be to change the `mtoShipment` value higher up in the component instead of in here to pull either `mtoShipment` or `mtoShipment.ppmShipment`

TevinAdams replied:
@danieljordan-caci I see what your saying but I do not think changing the actual value of mtoShipment would be good. I could add a ppmShipment variable and make things conditional off of a new variable but changing the mtoShipment variable I do not think would be best. I could move the conditional logic outside of the props though but not reassign the value of mtoShipment.

TevinAdams replied:
Yeah i agree. I can make that change.

TevinAdams replied:
Done

TevinAdams commented:
> > > @traskowskycaci This one is missing the originPostalCode, originCity and originState. In this case the header should not show up because it would be missing information. If you go to the DB and look at the shipment, populate those values and you will see the header show up.
> > 
> > 
> > Without modifying the db, how should I populate those values?
> 
> Only asking because I want to make sure we are actually populating those originPostalCode, originCity and originState values somewhere during the process and not just filling them out in the DB @TevinAdams

@traskowskycaci  The information for the header is sourced from the DB mto_shipments record -> addresses record. Those values are populated when the shipment is created. You could edit the shipment as a Counselor or a TOO though if not wanting to change it in the actual DB where the values for the header are pulled from.

danieljordan-caci commented on src/components/Office/ShipmentHeading/ShipmentHeading.jsx:
```diff
@@ -26,6 +26,12 @@ function ShipmentHeading({ shipmentInfo, handleShowCancellationModal, isMoveLock
+    shipmentInfo?.originState &&
+    shipmentInfo?.originPostalCode &&
+    shipmentInfo?.destinationAddress &&
+    shipmentInfo?.scheduledPickupDate;
```
This will always be false for `ppmShipment` objects - they do not have the `scheduledPickupDate` value in the object

danieljordan-caci replied:
Suggest you ask the gov which value they want to use for that

TevinAdams replied:
This was actually requested to be removed.

danieljordan-caci reviewed (changes requested):
Requesting changes

![Screenshot 2024-08-20 at 4 44 28 PM](https://github.com/user-attachments/assets/1b1ade7c-5897-49ed-b34c-f0bef6f94e83)

danieljordan-caci reviewed (approved):
LGTM. Agree with Maria's analysis and suggested fixes for the BL item.

TevinAdams commented:
> Would you mind updating the AC on the story to have the feature AC? BL AC is just "Given I am a TOO, TIO, Counselor" at the moment
> 
> Also requesting a quick screenshot of 508 compliance
> 
> Looks good testing it out locally so I will give you the green stamp as soon as those other things are addressed!

@traskowskycaci I was updating the ACs and must have jumped to something else and forgot to fix it it. Its fixed now. Removed the 508 compliance screenshot since the location date text was removed. But i added it back but its not going to show a check on that text because it was removed. I just included a 508 screenshot though.

traskowskycaci reviewed (approved):
Thanks for including that again and updating AC!
